### PR TITLE
style: format with Black and sort object lists

### PR DESCRIPTION
**Pull request by Justintime50**
* Styles the library with Black
* Adds the missing `.flake8` file
* Adds `pyproject.toml` file to help with Black and iSort formatting
* Adds Black and iSort checks to CI
* Sorts the object and prefixes lists alphabetically

### CODE REVIEW

**Review by jontsai — APPROVED** [review]
LGTM

**Comment by fiendish** [comment]
What black args are you setting? Black favors `"` for strings but I'm not seeing quotation marks being converted here.

**Comment by fiendish** [comment]
Also, how would people feel about adding a `black --check` step to the CI to enforce adherence?

**Review by fiendish — CHANGES_REQUESTED** [review]
(comments already added)

**Comment by Justintime50** [comment]
@fiendish yeah, on our team we prefer the single quotes.

The config I'm using is `--skip-string-normalization --line-length 120 --experimental-string-processing`. 

If we wanted to add a black formatting check on CI, we'd also want to add a black config file to the repo so it'll run the local black tool with the specific config we expect. I'll circle back to that eventually here, this is low priority compared to the other ongoing work. These are good thoughts!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -15,14 +15,19 @@ jobs:
...
       - name: install dependencies
-        run: python -m pip install flake8
-      - name: lint with flake8
+        run: python -m pip install -r requirements-dev.txt -e .
+      - name: Check lint with flake8
         run: flake8 --statistics easypost/ tests/ examples/
+      - name: Check formatting with Black
+        run: black easypost --check
+      - name: Check formatting with iSort
+        run: isort easypost --check-only
```
Comment (fiendish) [inline .github/workflows/ci.yml:24]: I think it's worth running these on the tests and examples directories as well

Comment (Justintime50) [reply .github/workflows/ci.yml]: Couldn't agree more, overlooked those - thanks!

**Review by jontsai — APPROVED** [review]

**Comment by Justintime50** [comment]
@Roguelazer anyway we could get the GitHub Action expected job of `flake8` changed to `lint` as we are now adding `Black` and `iSort` checks to that step and `flake8` no longer correctly identifies the job?

**Comment by fiendish** [comment]
> @Roguelazer anyway we could get the GitHub Action expected job of `flake8` changed to `lint` as we are now adding `Black` and `iSort` checks to that step and `flake8` no longer correctly identifies the job?

I believe this means just the name of the job.

**Pull request merged**